src/nvim/ex_cmds.h

```diff
@@ -31,30 +32,19 @@ typedef struct {
 
 // Defs for inc_sub functionality
 
-// List of matches in a line
-#define _noop(x)
-KLIST_INIT(colnr_T, colnr_T, _noop)
-typedef klist_t(colnr_T) klist_colnr_T;
-#undef _noop
+// List of matches on a single line
+typedef kvec_t(colnr_T) kvec_colnr_T;
 
 /// Structure to backup and display matched lines in incsubstitution
 typedef struct {
   linenr_T lnum;
   long nmatch;
   char_u *line;
-  klist_colnr_T *start_col;
+  kvec_colnr_T start_col;
 } MatchedLine;
 
 // List of matched lines
-#define _dealloc_MatchedLine(x) \
-do { \
-  if (x->data.line) { xfree(x->data.line); } \
-  if (x->data.start_col) { kl_destroy(colnr_T, x->data.start_col); } \
-  } while (0)
-
-KLIST_INIT(MatchedLine, MatchedLine, _dealloc_MatchedLine)
-typedef klist_t(MatchedLine) klist_MatchedLine;
-#undef _dealloc_MatchedLine
+typedef kvec_t(MatchedLine) kvec_MatchedLine;
```

Comment:
No I'm really bad at naming (ever heard of Leonardo of Quirm?), so I'm happy to use your suggestions. I obviously just contracted `kvec_t(colnr_T)` to something without brackets...